src/enclave/enclave_timer.c

```diff
@@ -28,7 +28,12 @@ _Atomic(uint64_t) internal_counter = 0;
  */
 uint64_t enclave_nanos()
 {
-    uint64_t e = sgxlkl_enclave_state.shared_memory.timer_dev_mem->nanos;
+    struct timer_dev* t = sgxlkl_enclave_state.shared_memory.timer_dev_mem;
+
+    if (!oe_is_outside_enclave(t, sizeof(struct timer_dev)))
```

Comment:
If the macros are moved into `enclave_util.h` they can be used here.

Reply:
Done